adapters/targetVideo/targetvideo.go

```diff
@@ -0,0 +1,194 @@
+		if len(request.Imp[i].Ext) == 0 {
+			continue
+		}
+		var root map[string]json.RawMessage
+		if err := json.Unmarshal(request.Imp[i].Ext, &root); err != nil {
+			// If ext cannot be parsed, skip transformation for this imp
+			continue
+		}
+
+		// Try to extract placementId from ext.bidder.targetVideo (or targetvideo)
+		placementId := ""
+		if bRaw, ok := root["bidder"]; ok && len(bRaw) > 0 {
+			var bidder map[string]json.RawMessage
+			if err := json.Unmarshal(bRaw, &bidder); err == nil {
+				if placementIdRaw, ok := bidder["placementId"]; ok && len(placementIdRaw) > 0 {
+
+					var asStr string
+					var asInt int64
+					if err := json.Unmarshal(placementIdRaw, &asStr); err == nil && asStr != "" {
+						placementId = asStr
+					} else if err := json.Unmarshal(placementIdRaw, &asInt); err == nil {
+						placementId = fmt.Sprintf("%d", asInt)
+					}
+
+				}
+			}
+			// Remove bidder node as required
+			delete(root, "bidder")
+		}
+
+		// If we obtained a placementId, set ext.prebid.storedrequest.id = placementId
+		if placementId != "" {
+			// Build prebid.storedrequest structure, preserving existing prebid if any
+			var prebid map[string]json.RawMessage
+			if pr, ok := root["prebid"]; ok && len(pr) > 0 {
+				_ = json.Unmarshal(pr, &prebid)
+			}
+			if prebid == nil {
+				prebid = make(map[string]json.RawMessage)
+			}
+			stored := map[string]string{"id": placementId}
+			storedRaw, _ := json.Marshal(stored)
+			prebid["storedrequest"] = storedRaw
+			prebidRaw, _ := json.Marshal(prebid)
+			root["prebid"] = prebidRaw
+		}
+
+		// Marshal back the transformed ext
+		if newExt, err := json.Marshal(root); err == nil {
+			request.Imp[i].Ext = newExt
+		}
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	//fmt.Println("TARGET VIDEO reqJson: ", string(reqJSON))
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}, nil
+}
+
+func (a *adapter) MakeBids(bidReq *openrtb2.BidRequest, unused *adapters.RequestData, httpRes *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if httpRes.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if httpRes.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", httpRes.StatusCode)}}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(httpRes.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{Message: fmt.Sprintf("error while decoding response, err: %s", err)}}
+	}
+
+	if len(bidResp.SeatBid) == 0 {
+		return nil, nil
+	}
+
+	if len(bidResp.SeatBid[0].Bid) == 0 {
+		return nil, nil
+	}
```

Comment:
You can delete these two checks because the `for range` on line 151 should gracefullly handle the empty scenarios. TODO: they are returning nil instead of an empty bidder response. Is this ok?